var/spack/repos/builtin/packages/tau/package.py

```diff
@@ -109,23 +171,80 @@ def install(self, spec, prefix):
         options = ["-prefix=%s" % prefix,
                    "-iowrapper",
                    "-pdt=%s" % spec['pdt'].prefix]
-        # If download is active, download and build suggested dependencies
-        if '+download' in spec:
-            options.extend(['-bfd=download',
-                            '-unwind=download',
-                            '-asmdex=download'])
-        else:
-            options.extend(["-bfd=%s" % spec['binutils'].prefix])
-            # TODO : unwind and asmdex are still missing
+#       If download is active, download and build suggested dependencies
+
+        options.extend(["-bfd=%s" % spec['binutils'].prefix])
+        options.extend(["-unwind=%s" % spec['libunwind'].prefix])
 
         if '+scorep' in spec:
             options.append("-scorep=%s" % spec['scorep'].prefix)
 
+        if '+likwid' in spec:
+            options.append("-likwid=%s" % spec['likwid'].prefix)
+
+        if '+papi' in spec:
+            options.append("-papi=%s" % spec['papi'].prefix)
+
+        if '+python' in spec:
+            options.append("-python")
+
         if '+openmp' in spec:
             options.append('-openmp')
 
+        if '+opari' in spec:
+            options.append('-opari')
+
         if '+mpi' in spec:
+            str_mpi_inc_tmp = ""
+            str_mpi_inc = ""
+            str_mpi_libs_tmp = ""
+            str_mpi_libs = ""
+            str_mpi_library_tmp = ""
+            str_mpi_library = ""
+            str_mpi_prefix = spec['mpi'].prefix
+            str_mpi = os.popen(str_mpi_prefix + '/bin/mpicc -show').read()
+            list_mpi_opts = str_mpi.split()
+
+            for mpi_item in list_mpi_opts:
+
+                if "-I" in mpi_item:
+                    if str_mpi_inc_tmp == "":
+                        str_mpi_inc_tmp = mpi_item[2:]
+                    else:
+                        str_mpi_inc_tmp += " "
+                        str_mpi_inc_tmp += mpi_item[2:]
+
+                    str_mpi_inc = str_mpi_inc_tmp
+
+                if "-L" in mpi_item:
+                    str_mpi_libs_tmp = mpi_item[2:]
+                    str_mpi_libs = str_mpi_libs_tmp
+
+                if "-l" in mpi_item:
+                    str_mpi_library_tmp = mpi_item
+
+                    str_mpi_library = str_mpi_library_tmp
+
             options.append('-mpi')
+            options.append('-mpiinc=' + str_mpi_inc)
+            options.append('-mpilib=' + str_mpi_libs)
+            libintl = spec['gettext'].prefix + '/lib'
+            options.append(
+                '-mpilibrary=' +
+                str_mpi_library +
+                ' -L' +
+                libintl +
+                ' -Wl,-rpath,' +
+                libintl + ' -lintl')
```

Comment:
This would benefit from either using either the [% operator](http://www.informit.com/articles/article.aspx?p=28790&seqNum=2) or [.format()](https://docs.python.org/3.4/library/string.html#format-examples).